packages/mars/src/options/config.ts

```diff
@@ -0,0 +1,83 @@
+import { providers, Wallet } from 'ethers'
+import Ganache from 'ganache-core'
+import { ExecuteOptions } from '../execute/execute'
+import { createJsonInputs } from '../verification'
+import { exit } from './checks'
+import { getCommandLineOptions } from './cli'
+import { getDefaultOptions } from './defaults'
+import { getEnvironmentOptions } from './environment'
+import { Options } from './Options'
+
+export async function getConfig(options: Options): Promise<ExecuteOptions> {
+  const merged = {
+    ...getDefaultOptions(),
+    ...getEnvironmentOptions(),
+    ...getCommandLineOptions(),
+    ...options,
+  }
+
+  if (merged.dryRun && merged.noConfirm === undefined) {
+    merged.noConfirm = true
+  }
+
+  let verification: ExecuteOptions['verification'] = undefined
+  if (merged.verify) {
+    verification = {
+      etherscanApiKey: merged.etherscanApiKey,
+      jsonInputs: await createJsonInputs(merged.sources),
+      waffleConfig: merged.waffleConfig,
+    }
+  }
+
+  const privateKey = merged.privateKey
+  if (privateKey === undefined) {
```

Comment:
I think we should leave option of running dry run w/o private key

Reply:
That is an excellent point!